.github/workflows/build.yml

```diff
@@ -0,0 +1,99 @@
+name: IntuneAppBuilder Deployment
+
+# TODO: Remove these comments once reviewed
+# on:
+#   schedules:
+#   - cron: "30 7 * * *"
+#     displayName: Daily job
+#     branches:
+#       include:
+#       - master
+#     always: true
+#   push:
+#     branches:
+#       - master
+#   pull_request:
+#     types: [opened, synchronize, reopened, closed]
+#     branches:
+#       - master
+
+on: workflow_dispatch
+
+env:
+  BuildConfiguration: 'Release'
+  Verbosity: 'normal'
+  ArtifactFile: 'output.nupkg'
+
+defaults:
+  runs-on: windows-latest
+  run:
+    shell: pwsh
+
+jobs:
+  Build:
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.301
+
+      - name: dotnet restore
+        run: dotnet restore --configuration $(Env:BuildConfiguration) --verbosity $(Env:Verbosity)
+      
+      - name: dotnet build
+        run: dotnet build --configuration $(Env:BuildConfiguration) --verbosity $(Env:Verbosity)
+
+      - name: dotnet pack
+        run: dotnet pack --configuration $(Env:BuildConfiguration) --verbosity $(Env:Verbosity) --no-build *.sln
+
+      - name: dotnet publish
+        run: dotnet publish '**/*/*.csproj' --configuration $(Env:BuildConfiguration) --no-build --verbosity $(Env:Verbosity) -o $(Env:ArtifactFile)
+
+      - name: dotnet test
+        run: dotnet test --configuration $(Env:BuildConfiguration) --no-build --verbosity $(Env:Verbosity)
+        env:
+          AadAuth:Password: ${{ secrets.TESTAADAUTHPASSWORD }}
+          AadAuth:Username: ${{ secrets.TESTAADAUTHUSERNAME }}
+
+      - name: Publish pipeline artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.ArtifactFile }}
+          path: ${{ env.ArtifactFile }}
+
+  DeployNuGet:
+    needs: Build
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.301
+
+      - name: Download Artifact from Previous Job
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ env.ArtifactFile }}
```

Comment:
again, there will be multiple artifacts, so can we download the whole artifacts directory?